src/ast/irbuilderbpf.cpp

```diff
@@ -830,6 +830,238 @@ CallInst *IRBuilderBPF::CreateGetNs(bool boot_time, const location &loc)
+
+  CreateBr(arr_ne);
+  SetInsertPoint(arr_ne);
+  Value *result = CreateLoad(getInt1Ty(), store);
+  CreateLifetimeEnd(store);
+  CreateLifetimeEnd(v1);
+  CreateLifetimeEnd(v2);
+  result = CreateIntCast(result, getInt64Ty(), false);
+  return result;
+}
+
+Value *IRBuilderBPF::CreateIntegerArrayCmp(Value *ctx,
+                                           Value *val1,
+                                           Value *val2,
+                                           const SizedType &val1_type,
+                                           const SizedType &val2_type,
+                                           const bool inverse,
+                                           const location &loc,
+                                           MDNode *metadata)
+{
+  /*
+   // This function compares each character of the two arrays.
+   // It returns true if all are equal and false if any are different
+   // cmp([]char val1, []char val2)
+   {
+      for (size_t i = 0; i < n; i++)
+      {
+        if (val1[i] != val2[i])
+        {
+          return false;
+        }
+      }
+      return true;
+   }
+*/
+  auto elem_type = *val1_type.GetElementTy();
+  const size_t num = val1_type.GetNumElements();
+
+  Value *val1_elem_i, *val2_elem_i, *cmp;
+  AllocaInst *v1 = CreateAllocaBPF(elem_type, "v1");
+  AllocaInst *v2 = CreateAllocaBPF(elem_type, "v2");
+  AllocaInst *store = CreateAllocaBPF(getInt1Ty(), "arraycmp.result");
+  CreateStore(getInt1(!inverse), store);
+
+  Function *parent = GetInsertBlock()->getParent();
+  BasicBlock *while_cond = BasicBlock::Create(module_.getContext(),
+                                              "while_cond",
+                                              parent);
+  BasicBlock *while_body = BasicBlock::Create(module_.getContext(),
+                                              "while_body",
+                                              parent);
+  BasicBlock *while_end = BasicBlock::Create(module_.getContext(),
+                                             "while_end",
+                                             parent);
+  BasicBlock *arr_ne = BasicBlock::Create(module_.getContext(),
+                                          "arraycmp.false",
+                                          parent);
+  BasicBlock *done = BasicBlock::Create(module_.getContext(),
+                                        "arraycmp.done",
+                                        parent);
+
+  Value *ptr_val1 = CreateIntToPtr(val1, GetType(val1_type)->getPointerTo());
+  Value *ptr_val2 = CreateIntToPtr(val2, GetType(val2_type)->getPointerTo());
+  AllocaInst *i = CreateAllocaBPF(getInt32Ty(), "i");
+  AllocaInst *n = CreateAllocaBPF(getInt32Ty(), "n");
+  CreateStore(getInt32(0), i);
+  CreateStore(getInt32(num), n);
+
+  CreateBr(while_cond);
+  SetInsertPoint(while_cond);
+  auto *cond = CreateICmpSLT(CreateLoad(getInt32Ty(), i),
+                             CreateLoad(getInt32Ty(), n),
+                             "true_cond");
```

Comment:
Maybe
```suggestion
                             "size_check");
```
`true_cond` doesn't seem like a good name.